# DataFrame Analysis Exercise

**Import the `bestsellers.csv` dataset and use it to answer the following questions:**

* Find the lowest User Rating in the DF
* Find the highest Price in the DF
* What is the average User Rating?
* What the average User Rating of the first 5 rows?
* What User Review score appeared the most?
* What is the total (sum) of all the values in the Reviews column?
* How many different authors are featured in the dataset?
* Which author wrote the most number of books on the list?  How many did they write?

In [1]:
import pandas as pd

In [2]:
bestseller = pd.read_csv("data/bestsellers.csv")

In [3]:
bestseller.head()

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [4]:
bestseller.min()

Name           10-Day Green Smoothie Cleanse
Author                      Abraham Verghese
User Rating                              3.3
Reviews                                   37
Price                                      0
Year                                    2009
Genre                                Fiction
dtype: object

En düşük User Rating değeri = 3.3

In [5]:
bestseller.max()

Name           You Are a Badass: How to Stop Doubting Your Gr...
Author                                              Zhi Gang Sha
User Rating                                                  4.9
Reviews                                                    87841
Price                                                        105
Year                                                        2019
Genre                                                Non Fiction
dtype: object

En yüksek Price değeri = 105

In [6]:
# Ortalama User Rating değeri
bestseller["User Rating"].mean()

4.618363636363641

In [7]:
bestseller[0:6].mean(numeric_only=True)

User Rating        4.650000
Reviews        13352.166667
Price             12.333333
Year            2015.333333
dtype: float64

In [8]:
bestseller.head().mean(numeric_only=True)

User Rating        4.7
Reviews        13494.0
Price             12.6
Year            2016.2
dtype: float64

İlk beş satır için ortalama User Rating değeri = 4.7

In [9]:
bestseller.mode(numeric_only=True)

User Rating  Reviews  Price  Year
0           4.8   8580.0    8.0  2009
1           NaN      NaN    NaN  2010
2           NaN      NaN    NaN  2011
3           NaN      NaN    NaN  2012
4           NaN      NaN    NaN  2013
5           NaN      NaN    NaN  2014
6           NaN      NaN    NaN  2015
7           NaN      NaN    NaN  2016
8           NaN      NaN    NaN  2017
9           NaN      NaN    NaN  2018
10          NaN      NaN    NaN  2019

En çok tekrar eden Review değeri = 4.8

In [10]:
# Reviews sütunun tuttuğu değerlerin toplamı
bestseller["Reviews"].sum()

6574305

In [11]:
bestseller.describe(include="O")

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   

              Genre  
count           550  
unique            2  
top     Non Fiction  
freq            310

<br>Veri seti içerisinde birbirinden farklı 248 yazar bulunmaktadır.<br>
Veri seti içerisinde en çok kitap yazan yazar, 12 kitap ile Jeff Kinney'dir